src/engine/IdTable.h

```diff
@@ -23,151 +23,44 @@ class IdTableImpl {
-        // If we manage our own storage recreate that to fit the other row
-        delete[] _data;
-        if (other._allocated) {
-          // Both rows manage their own storage so we can take advantage
-          // of move semantics.
-          _data = other._data;
-          return *this;
-        } else {
-          _data = new Id[COLS];
-        }
-      }
-      // Copy over the data from the other row to this row
-      if (_data != nullptr && other._data != nullptr) {
-        std::memcpy(_data, other._data, sizeof(Id) * COLS);
-      }
-      return *this;
-    }
-
-    bool operator==(const Row& other) const {
-      bool matches = true;
-      for (size_t i = 0; matches && i < COLS; i++) {
-        matches &= _data[i] == other._data[i];
-      }
-      return matches;
-    }
-
-    Id& operator[](size_t i) { return *(_data + i); }
-    const Id& operator[](size_t i) const { return *(_data + i); }
-    Id* data() { return _data; }
-    const Id* data() const { return _data; }
-
-    size_t size() const { return COLS; }
-    size_t cols() const { return COLS; }
-
-    Id* _data;
-    bool _allocated;
-  };
-
-  friend std::ostream& operator<<(std::ostream& out,
-                                  const typename IdTableImpl<COLS>::Row& row) {
-    for (size_t col = 0; col < row.size(); col++) {
-      out << row[col] << ", ";
-    }
-    out << std::endl;
-    return out;
-  }
-
-  friend std::ostream& operator<<(
-      std::ostream& out, const typename IdTableImpl<COLS>::ConstRow&& row) {
-    for (size_t col = 0; col < row.size(); col++) {
-      out << row[col] << ", ";
-    }
-    out << std::endl;
-    return out;
-  }
+  //   friend std::ostream& operator<<(std::ostream& out,
+  //                                   const typename IdTableImpl<COLS>::Row&
+  //                                   row) {
+  //     for (size_t col = 0; col < row.size(); col++) {
+  //       out << row[col] << ", ";
+  //     }
+  //     out << std::endl;
+  //     return out;
+  //   }
+  //
+  //   friend std::ostream& operator<<(
+  //       std::ostream& out, const typename IdTableImpl<COLS>::ConstRow&& row)
+  //       {
+  //     for (size_t col = 0; col < row.size(); col++) {
+  //       out << row[col] << ", ";
+  //     }
+  //     out << std::endl;
+  //     return out;
+  //   }
 
   class iterator {
    public:
     // iterator traits types
     using difference_type = ssize_t;
-    using value_type = Row;
-    using pointer = Row*;
-    using reference = Row&;
+    using value_type = std::array<Id, COLS>;
+    using pointer = value_type*;
+    using reference = value_type&;
     using iterator_category = std::forward_iterator_tag;
 
-    iterator() : _data(nullptr), _row(0), _rowView(nullptr, 0) {}
-    iterator(Id* data, size_t row, size_t cols)
-        : _data(data), _row(row), _rowView(data + (cols * row), cols) {}
+    iterator() : _data(nullptr), _row(0) {}
+    iterator(Id* data, size_t row, size_t) : _data(data), _row(row) {}
```

Comment:
The iterator has to take the number of columns as a parameter to allow for compatibility between the fixed width and variable with `IdTable`. To prevent the compiler from complaining about an unused parameter we can either not give the parameter a name, or cast it to void in the constructors body (e.g. `(void)cols;`).
I've added a comment explaining the unnamed parameter.

Reply:
you could also use
```
iterator(Id* data, size_t row, size_t  /* cols */) : _data(data), _row(row) {}
```